<?xml version="1.0" ?>

<launch>
  <arg name="gripper_name" default="gripper" />
  <arg name="min_gap" default="-0.017" />
  <arg name="max_gap" default="0.085" /> <!-- meters -->
  <arg name="min_effort" default="30.0" />
  <arg name="max_effort" default="100.0" />
  <arg name="serial_port" default="/dev/ttyUSB0" />

  <group ns="$(arg gripper_name)">
    <node name="Robotiq2FGripperRtuNode" pkg="robotiq_2f_gripper_control"
      type="Robotiq2FGripperRtuNode.py" args="$(arg serial_port)">
      <remap from="Robotiq2FGripperRobotInput" to="input" />
      <remap from="Robotiq2FGripperRobotOutput" to="output" />
    </node>
  </group>

  <node name="robotiq_2f_gripper_action_server" pkg="robotiq_2f_gripper_action_server"
    type="robotiq_2f_gripper_action_server_node" output="screen">
    <param name="gripper_name" type="str" value="$(arg gripper_name)" />
    <param name="min_gap" type="double" value="$(arg min_gap)" />
    <param name="max_gap" type="double" value="$(arg max_gap)" />
    <param name="min_effort" type="double" value="$(arg min_effort)" />
    <param name="max_effort" type="double" value="$(arg max_effort)" />

    <remap from="input" to="$(arg gripper_name)/input" />
    <remap from="output" to="$(arg gripper_name)/output" />
  </node>

  <node name="robotiq_2f_client" pkg="robotiq_2f_gripper_serial_control" type="robotiq_2f_client.py" output="screen">
  </node>

</launch>
